primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java

```diff
@@ -220,22 +221,24 @@ protected Predicate createPredicate(FilterMeta filter, Field filterField,
         Lazy<Collection<Object>> filterValueAsCollection = new Lazy(
                 () -> filterValue.getClass().isArray() ? Arrays.asList((Object[]) filterValue)
                         : (Collection<Object>) filterValue);
+        FacesContext context = FacesContext.getCurrentInstance();
```

Comment:
createPredicate is also called in a loop

Reply:
Moved it up higher outside the loops.